tracing/opentelemetry/opentelemetry.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package opentelemetry
```

Comment:
Can be simplified to `otel` if we want:

```suggestion
package otel
```

Reply:
Let's keep naming consistent with already existing parts (`opentracing` and `opentelemetry`).